### PR TITLE
[SE-0289] Update the detailed design, etc.

### PR DESCRIPTION
**Pull request by benrimmington**

### CODE REVIEW

File: proposals/0289-result-builders.md [hunk 1]
```diff
@@ -180,7 +180,7 @@ header1(chapter + "1. Loomings.")
...
+Most programmers would advise declaring this variable in as narrow a scope as possible, and as close as possible to where it's going to be used.  But because the entire hierarchy is an expression, and there's no easy way to declare variables within expressions, every variable like this has to be declared above the entire hierarchy.  (Now, it's true that there's a trick for declaring locals within expressions: you can start a closure, which gives you a local scope that you can use to declare whatever you want, and then immediately call it. But this is awkward in its own way and significantly adds to the punctuation problem.)
```
Comment (benrimmington) [inline proposals/0289-result-builders.md:183]: * Added comma after `possible`
* Added `way` after `no easy`

File: proposals/0289-result-builders.md [hunk 2]
```diff
@@ -1118,7 +1121,7 @@ Support for additional control-flow statements would weaken the declarative feel
...
+It is common for DSLs to want to introduce shorthands which might be unreasonable to introduce into the global scope.  For example, `p` might be a reasonable name in the context of our `HTMLBuilder` DSL (rather than `paragraph`), but actually introducing a global function named `p` just for DSL use is quite unfortunate.  Contextual lookups like `.p` will generally not work at the top level in DSLs because they will be interpreted as continuations of the previous statement. One could imagine having some way for the DSL to affect lexical lookup within transformed functions so that, e.g., within the transformed function one could use short names like `p`, `div`, and `h1`:
```
Comment (benrimmington) [inline proposals/0289-result-builders.md:1124]: * Removed `not` before `be unreasonable`

File: proposals/0289-result-builders.md [hunk 3]
```diff
@@ -272,9 +274,9 @@ To be useful as a result builder, the result builder type must provide a suffici
...
+This is a quick reference for the result-building methods currently proposed.  The typing here is subtle, as it often is in macro-like features.  In the following descriptions, `Expression` stands for any type that is acceptable for an expression-statement to have (that is, a raw partial result), `Component` stands for any type that is acceptable for a partial or combined result to have, and `FinalResult` stands for any type that is acceptable to be ultimately returned by the transformed function.
 
-* `buildBlock(_ components: Component...) -> Component` is used to build combined results for statement blocks. It is required to be a static method in every result builder.
+* `buildBlock(_ components: Component...) -> Component` is used to build combined results for statement blocks. It is required to be a static method in every result builder. It can have zero/one/many/variadic parameters.
```
Comment (benrimmington) [inline proposals/0289-result-builders.md:279]: Or would it be better to add this to the HTMLBuilder example:

```swift
  // We can provide this overload as another micro-optimization.
  static func buildBlock() -> Component {
    return []
  }
```

**Comment by benrimmington** [comment]
@DougGregor Do you want to merge any of these changes now, in case you decide to rename the attribute?

**Review by DougGregor — APPROVED** [review]
These look good, thank you!

**Pull request merged**